lib/src/utils.ts

```diff
@@ -34,6 +34,19 @@ export function isOmnilockScript(script: Script, config: SporeConfig) {
   return script.codeHash === omnilockScript.CODE_HASH && script.hashType === omnilockScript.HASH_TYPE;
 }
 
+export function ScriptEqualDeepCheck(script_a: Script, script_b: Script, config: SporeConfig) {
+  const omni_a = isOmnilockScript(script_a, config);
+  const omni_b = isOmnilockScript(script_b, config);
+  if (omni_a !== omni_b) {
+    return false;
+  }
+  if (omni_a && omni_b) {
+    return script_a.args.slice(0, 42) == script_b.args.slice(0, 42);
+  } else {
+    return script_a === script_b;
+  }
```

Comment:
can be refactored in more clear

```ts
if (!bytes.equal(Script.pack(script_a), Script.pack(script_b))) {
  return false
}

return isOmnilockScript(script_a, config)
```

Reply:
this method to to compare two different Omni script, which are Omni-ACP lock and Omni default lock that share a same public key, so we cannot directly compare them in bytes

Reply:
I see